pkg/lib/config/saml.go

```diff
@@ -204,3 +206,64 @@ type SAMLSigningDigestMethod string
 const (
 	SAMLSigningDigestMethodSHA256 SAMLSigningDigestMethod = "SHA256"
 )
+
+var _ = Schema.Add("SAMLAttributesConfig", `
+{
+	"type": "object",
+	"additionalProperties": false,
+	"properties": {
+		"definitions": {
+			"type": "array",
+			"items": { "$ref": "#/$defs/SAMLAttributeDefinition" }
+		}
+	}
+}
+`)
+
+type SAMLAttributesConfig struct {
+	Definitions []SAMLAttributeDefinition `json:"definitions,omitempty"`
+}
+
+var _ = Schema.Add("SAMLAttributeDefinition", `
+{
+	"type": "object",
+	"additionalProperties": false,
+	"properties": {
+		"name": { "type": "string" },
+		"name_format": { "type": "string" },
+		"friendly_name": { "type": "string" }
+	},
+	"required": ["name"]
+}
+`)
+
+type SAMLAttributeDefinition struct {
+	Name         string                  `json:"name,omitempty"`
+	NameFormat   SAMLAttributeNameFormat `json:"name_format,omitempty"`
+	FriendlyName string                  `json:"friendly_name,omitempty"`
+}
+
+func (c *SAMLAttributeDefinition) SetDefaults() {
```

Comment:
I forget whether a struct in a slice would have its SetDefaults() called. Let's add a test to verify this.

Or you can rename NameFormat to NameFormat_WriteOnly and add a GetNameFormat getter.

Reply:
I've tested it will be called. Let me add a test for it.